ts/features/newWallet/components/WalletCardsCategoryContainer.tsx

```diff
@@ -4,28 +4,33 @@ import {
   VSpacer,
   WithTestID
 } from "@pagopa/io-app-design-system";
+import { identity } from "lodash";
 import * as React from "react";
 import Animated, { Layout } from "react-native-reanimated";
 import { WalletCard } from "../types";
 import { renderWalletCardFn } from "../utils";
+import { WalletCardSkeleton } from "./WalletCardSkeleton";
 
-export type WalletCategoryStackContainerProps = WithTestID<{
+type BaseCategoryContainerProps<T> = WithTestID<{
   iconName: IOIcons;
   label: string;
-  cards: ReadonlyArray<WalletCard>;
+  cards: ReadonlyArray<T>;
 }>;
 
-/**
- * This component handles the rendering of cards of a specific category.
- * The component also handles logic behind card stacking and animations
- */
-const WalletCardsCategoryContainer = ({
+export type RenderProps<T> = {
+  keyExtractorFn: (card: T) => string;
+  renderCardnFn: (card: T, isStacked: boolean) => JSX.Element | null;
+};
+
+const BaseCategoryContainer = <T,>({
+  testID,
   label,
   iconName,
   cards,
-  testID
-}: WalletCategoryStackContainerProps) => {
-  if (cards === undefined || cards.length === 0) {
+  keyExtractorFn,
+  renderCardnFn
```

Comment:
Is it the right name or a typo?

Reply:
Oops, there's one too many "n". Fixed here: https://github.com/pagopa/io-app/pull/5707/commits/09810fda6c2a319e2f5f884cd858602fea9745b6